emcc.py

```diff
@@ -562,6 +562,10 @@ def get_binaryen_passes():
     if settings.LEGALIZE_JS_FFI:
       # legalize it again now, as the instrumentation may need it
       passes += ['--legalize-js-interface']
+      if settings.RELOCATABLE:
+        passes += ['--pass-arg=legalize-js-interface-export-originals']
+      if not settings.SIDE_MODULE:
+        passes += ['--pass-arg=legalize-js-interface-exported-helpers']
   if settings.EMULATE_FUNCTION_POINTER_CASTS:
```

Comment:
How does this work for a side module - is it exporting the helpers from the side module to the outside? I'd expect the reverse so I'm not sure I follow.

Reply:
Side modules remain unchanged in that they continue to import these helpers.  The helpers are defined in compiler-rt which is always/only linked into the main module.  The main module exports them and side modules import them.